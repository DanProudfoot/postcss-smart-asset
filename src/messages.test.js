import path from "path"
import postcss from "postcss"
import postcssUrl from "."
import { read } from "../test/setup"

describe("misc", () => {
  test("should add dependency messages with copy", () => {
    return postcss()
      .use(
        postcssUrl({
          url: "copy",
          assetsPath: "test/fixtures/build/assets"
        })
      )
      .process(read("copy"), {
        from: "test/fixtures/copy.css"
      })
      .then((result) => {
        const dependencies = result.messages.filter((m) => m.type === "dependency")
        expect(dependencies).toMatchSnapshot()
      })
  })

  test("should add dependency messages with inline", () => {
    return postcss()
      .use(
        postcssUrl({
          url: "inline",
          assetsPath: "test/fixtures/build/assets"
        })
      )
      .process(read("copy"), {
        from: "test/fixtures/copy.css"
      })
      .then((result) => {
        const dependencies = result.messages.filter((m) => m.type === "dependency")
        expect(dependencies).toMatchSnapshot()
      })
  })
})
