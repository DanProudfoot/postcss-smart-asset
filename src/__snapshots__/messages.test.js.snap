// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`misc should add dependency messages with copy 1`] = `
Array [
  Object {
    "file": "/Users/sebastian/Workspace/sebastian-software/postcss-smart-asset/test/fixtures/imported/pixel.png",
    "parent": "/Users/sebastian/Workspace/sebastian-software/postcss-smart-asset/test/fixtures/copy.css",
    "type": "dependency",
  },
  Object {
    "file": "/Users/sebastian/Workspace/sebastian-software/postcss-smart-asset/test/fixtures/pixel.gif",
    "parent": "/Users/sebastian/Workspace/sebastian-software/postcss-smart-asset/test/fixtures/copy.css",
    "type": "dependency",
  },
]
`;

exports[`misc should add dependency messages with inline 1`] = `
Array [
  Object {
    "file": "/Users/sebastian/Workspace/sebastian-software/postcss-smart-asset/test/fixtures/imported/pixel.png",
    "parent": "/Users/sebastian/Workspace/sebastian-software/postcss-smart-asset/test/fixtures/copy.css",
    "type": "dependency",
  },
  Object {
    "file": "/Users/sebastian/Workspace/sebastian-software/postcss-smart-asset/test/fixtures/pixel.gif",
    "parent": "/Users/sebastian/Workspace/sebastian-software/postcss-smart-asset/test/fixtures/copy.css",
    "type": "dependency",
  },
]
`;
